compiler/rustc_codegen_llvm/src/llvm_util.rs

```diff
@@ -218,13 +218,34 @@ pub fn target_cpu(sess: &Session) -> &str {
     handle_native(name)
 }
 
-pub fn handle_native_features(sess: &Session) -> Vec<String> {
+/// The list of LLVM features computed from CLI flags (`-Ctarget-cpu`, `-Ctarget-feature`,
+/// `--target` and similar).
+// FIXME(nagisa): Cache the output of this somehow? Maybe make this a query? We're calling this
+// for every function that has `#[target_feature]` on it. The global features won't change between
+// the functions; only crates, maybe…
+pub fn llvm_global_features(sess: &Session) -> Vec<String> {
+    // FIXME: this should definitely be available more centrally and to other codegen backends.
+    /// These features control behaviour of rustc rather than llvm.
+    const RUSTC_SPECIFIC_FEATURES: &[&str] = &["crt-static"];
+
+    // Features that come earlier are overriden by conflicting features later in the string.
+    // Typically we'll want more ecplicit settings to override the implicit ones, so:
+    //
+    // * Features implied by --target; are overriden by
+    // * Features from -Ctarget-cpu=native; are overriden by
```

Comment:
How does `-Ctarget-cpu=something-specific` fit into this picture?

Reply:
Hm, that's a good question. Today we only pass a computed list of `target-feature`s for `-Ctarget-cpu=native` only.

For other `-Ctarget-cpu`s the `target-feature` list we pass to LLVM will only contain values from `--target` and `-Ctarget-feature` flag. The features present in the string we pass to LLVM will override those that are implied from `-Ctarget-cpu` (regardless of whether it is set implicitly through target or explicitly via CLI).

In short, there ends up being a difference in feature priority: `-Ctarget-cpu=!native` is overriden by `target-features` implied by `--target` is overriden by `-Ctarget-cpu=native` is overriden by `-Ctarget-features`.

I think we still want `-Ctarget-cpu` to override the features implied by `--target`, but that sounds difficult to handle in all cases (including the one where no `-Ctarget-cpu` flag is specified at all...)

I'll sleep on this.

Reply:
I'm going to change it back to the previous ordering (`-Ctarget-cpu` -> `--target` -> `-Ctarget-features`) just to facilitate landing this sooner and figure out something about the ordering otherwise later.